### DISCUSSION TITLE
PNP People Search Omit Fields being searched

### DISCUSSION THREAD

**Discussion by StephenHoughton4** [microsoft-search/pnp-modern-search#2985] [Q&A]
Hi,

So I have a query where there is an issue with the results provided from my People Search. If I search "FAL", I get Jimmy Fallon and Davy Jones. The only reason Davy Jones is returned is because he is in an AD Group called "Falcon_Trainers". This sort of thing is polluting the results section.

I tried editing the Search Query, but this led to another issue, so any help would be appreciated. My Current Search query (plus the part I added to the end on the row below)

{SearchTerms}*
-Memberships:{SearchTerms}*

The addition worked on the surface (it removes Davy Jones), but there is an issue, If I add Jimmy Fallon to the Falcon_Trainers group, the search would produce no results.

Is there a way to omit the Memberships from being imported or at least not included when searching?

I have a previous question (separate issue - [2922](https://github.com/microsoft-search/pnp-modern-search/discussions/2922) ), in which, I queried the cause of spurious results, but this is to work on removing them.

Any help appreciated.

Thanks
Stephen

**Comment by kasperbolarsen** [comment]
I am using one of two options with my clients.  Either only search for the query text in specific properties ( name, phone, skills,  prior projects etc) or search in all properties and using XRANK to ensure that the most relevant shows up on top.

**Reply by StephenHoughton4** [reply]
@kasperbolarsen  Would you be able to use [Project-Away](https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/projectawayoperator) to omit the results form appearing? If so, would you know where to put this?

I tried previously to search only certain columns, but If I search "FAL USA" (2 words for 2 columns), I would get no results. Also It is unfeasible to ask users to have to type "FAL Office:USA" As they would all need training on the searchable attributes and KQL, so a user doesn't have to search "USA LastName:Fal*" to get that result to appear.

I am currently using the XRANK Suggestion you supplied, but hoping to remove the results that don't actually appear to match to the general users).

Thanks for your input so far, I have amended the Query to as follows:
`{?(((((({searchTerms}* 
XRANK(stdb=100000) PersonInitials:{searchTerms}*) 
XRANK(stdb=90000) LastName:{searchTerms}*) 
XRANK(stdb=80000) FirstName:{searchTerms}*)  
XRANK(stdb=70000) Office:{searchTerms}*)  
XRANK(stdb=60000) JobTitle:{searchTerms}*)
XRANK(stdb=-10000) Memberships:{searchTerms}*)
} `

**Reply by wobba** [reply]
Those values for stdb are just WOW! Just saying 🙂

Also, if you have multiple terms use () on the properties like:

`FirstName:({searchTerms}*)`

See https://learn.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference#grouping-property-restrictions-within-a-kql-query
